src/System.IO.FileSystem/src/System/IO/UnixFileStream.cs

```diff
@@ -1284,24 +1284,16 @@ private long SeekCore(long offset, SeekOrigin origin)
             Debug.Assert(handle != null);
             Debug.Assert(!handle.IsInvalid);
 
-            // System calls may fail due to EINTR (signal interruption).  We need to retry in those cases.
-            while (true)
+            long result = sysCall(handle, arg1, arg2);
```

Comment:
Similarly, now that there's no longer a loop that requires potentially invoking sysCall multiple times, this can be refactored such that we no longer need to pass a delegate around.  It'll make it cleaner and lead to less overheads.